### PR TITLE
feat: add a toggle project centroids component

### PR DESCRIPTION
**Pull request by steveoh**
this component allows the features to be toggled with the centroids. The lock icon locks in the choice regardless of the scale.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -45,5 +45,17 @@
...
+    "*.ts": "${capture}.js",
+    "*.js": "${capture}.js.map, ${capture}.min.js, ${capture}.d.ts",
+    "*.jsx": "${capture}.js",
+    "*.tsx": "${capture}.ts, ${capture}.stories.tsx",
+    "tsconfig.json": "tsconfig.*.json",
+    "package.json": "package-lock.json, yarn.lock, pnpm-lock.yaml, bun.lockb",
+    ".env": ".env.local, .env.development, .env.production, .env.test, .env.staging, .env.local.example, .env.development.example, .env.production.example, .env.test.example, .env.staging.example",
+    "firebase.json": ".firebaserc, firestore.indexes.json, firestore.rules, storage.rules",
+    ".prettierrc": ".prettierignore"
+  }
```
Comment (stdavis) [inline .vscode/settings.json:60]: Did you mean to put these into this project's settings? They seem like personal preferences to me.

**Review by stdavis — CHANGES_REQUESTED** [review]
The code looks good to me. However, it appears that the preview is 404ing on some bundles.

File: .vscode/settings.json (hunk 1 above)
Comment (steveoh) [reply .vscode/settings.json:60]: I noticed today you had these in your project and was envious. Now that I look closer at electrofishing, you must have them as a user preference. I'll pull it out.

**Comment by steveoh** [comment]
> it appears that the preview is 404ing on some bundles.

Hmm, the prod website will be hosted on the wri slash so it will work in prod. I'll need to find a solution to allow the previews to continue to function.

File: .vscode/settings.json (hunk 1 above)
Comment (stdavis) [reply .vscode/settings.json]: If nothing else, it would dry up the config files for our individual projects.

**Comment by steveoh** [comment]
Ok, the preview should be good to go.

**Review by stdavis — APPROVED** [review]

**Pull request merged**